### PR TITLE
RolePicker: Use portal to render menu

### PR DESCRIPTION
**Pull request by alexanderzobnin**
**What is this feature?**

Render role picker menu with portal.

**Why do we need this feature?**

This improves role picker behaviour inside other components (prevents stretch/scrolling)

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-enterprise/issues/5949

**Special notes for your reviewer:**

### CODE REVIEW

File: public/app/core/components/RolePicker/RolePicker.tsx [hunk 1]
```diff
@@ -82,14 +82,13 @@ export const RolePicker = ({
      * both (the role picker menu and its sub menu) aligned to the left edge of the input.
      * Otherwise, it aligns the role picker menu to the right.
      */
-    if (
-      window.innerWidth - right < currentRolePickerWidth &&
-      currentRolePickerWidth < 2 * ROLE_PICKER_SUBMENU_MIN_WIDTH
-    ) {
-      horizontal = offsetHorizontal;
+    if (left + ROLE_PICKER_MAX_MENU_WIDTH > window.innerWidth) {
+      horizontal = window.innerWidth - right;
```
Comment (alexanderzobnin) [inline public/app/core/components/RolePicker/RolePicker.tsx:86]: I push menu to the right side of input to get a bit more space at the left (it's also easier to calculate).

**Comment by alexanderzobnin** [comment]
Unfortunately, I've `<ClickOutsideWrapper>` does not work correctly with portaled menu since it's rendered outside a component DOM tree. I need to find a way to figure this out.

**Comment by Clarity-89** [comment]
> Unfortunately, I've `<ClickOutsideWrapper>` does not work correctly with portaled menu since it's rendered outside a component DOM tree. I need to find a way to figure this out. I think I can try to use `useOverlay` from `@react-aria/overlays`.

Maybe looking at how our Select component does it would be a good starting point.

**Comment by alexanderzobnin** [comment]
Ok, seems simple trick with `stopPropagation()` works in this case.

**Review by Clarity-89 — COMMENTED** [review]
Works great! With this PR we'd also remove the `TableWrapper` component as it's not necessary anymore.

**Comment by alexanderzobnin** [comment]
I noticed some strange bug where on wide screen menu populated at the top left corner. Need to fix it.

**Comment by alexanderzobnin** [comment]
@Clarity-89 I would prefer to remove it in a separate PR

**Review by Clarity-89 — APPROVED** [review]

**Pull request merged**